### PR TITLE
[DependencyInjection] non-conflicting anonymous service ids across files

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 3.3
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #23483
| License       | MIT
| Doc PR        |

### CODE REVIEW

**Review by ro0NL — COMMENTED** [review]
XML loader has the same issue?

File: src/Symfony/Component/DependencyInjection/Tests/Loader/YamlFileLoaderTest.php [hunk 1]
```diff
@@ -546,7 +546,7 @@ public function testAnonymousServices()
         $this->assertInternalType('array', $factory);
         $this->assertInstanceOf(Reference::class, $factory[0]);
         $this->assertTrue($container->has((string) $factory[0]));
-        $this->assertStringStartsWith('1', (string) $factory[0]);
+        $this->assertRegExp('/^\d+_[A-Z0-9]{64}$/i', (string) $factory[0]);
```
Comment (ro0NL) [inline src/Symfony/Component/DependencyInjection/Tests/Loader/YamlFileLoaderTest.php]: Coul add `a-z` range to avoid `i` modifier

**Comment by xabbuh** [comment]
No, the `XmlFileLoader` use the absolute path of the XML files when generating the internal id. But that made me think we should probably just do the same here (code updated accordingly).

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**